Spec/Auth.swift

```diff
@@ -205,6 +205,272 @@ class Auth : QuickSpec {
+                            let options = AblyTests.clientOptions()
+                            options.authCallback = { tokenParams, completion in
+                                completion(nil, NSError(domain: NSURLErrorDomain, code: -1003, userInfo: [NSLocalizedDescriptionKey: "A server with the specified hostname could not be found."]))
+                            }
+                            let realtime = ARTRealtime(options: options)
+                            defer { realtime.dispose(); realtime.close() }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                realtime.connection.once(.Disconnected) { stateChange in
+                                    guard let stateChange = stateChange else {
+                                        fail("ConnectionStateChange is nil"); done(); return
+                                    }
+                                    guard let errorInfo = stateChange.reason else {
+                                        fail("ErrorInfo is nil"); done(); return
+                                    }
+                                    expect(errorInfo.code) == 80019
+                                    done()
+                                }
+                            }
+
+                            guard let errorInfo = realtime.connection.errorReason else {
+                                fail("ErrorInfo is empty"); return
+                            }
+                            expect(errorInfo.code) == 80019
+                        }
+
+                        // RSA4c3
+                        it("if the connection is CONNECTED, then the connection should remain CONNECTED") {
+                            let options = AblyTests.clientOptions()
+                            options.authCallback = { tokenParams, completion in
+                                completion(getTestTokenDetails(ttl: 2.0), nil)
+                            }
+                            let realtime = ARTRealtime(options: options)
+                            defer { realtime.dispose(); realtime.close() }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                realtime.connection.once(.Connected) { stateChange in
+                                    expect(stateChange?.reason).to(beNil())
+                                    done()
+                                }
+                            }
+
+                            // Token should renew and fail
+                            realtime.options.authCallback = { tokenParams, completion in
+                                completion(nil, NSError(domain: NSURLErrorDomain, code: -1003, userInfo: [NSLocalizedDescriptionKey: "A server with the specified hostname could not be found."]))
+                            }
+
+                            expect(realtime.connection.errorReason).toEventuallyNot(beNil(), timeout: testTimeout)
```

Comment:
As @mattheworiordan says above, this should happen because of an incoming AUTH and not because the token expires, because the token expiring, as per [RTN15a](http://docs.ably.io/client-lib-development-guide/features/#RTN15a), causes a transition to DISCONNECTED right away instead of staying CONNECTED.